### DISCUSSION TITLE
Stoi_loss problem with enhancement template

### DISCUSSION THREAD

**Discussion by MaizeKai** [speechbrain/speechbrain#1776] [Q&A]
Hello, I use the enhancement template you provide to train my own model with google speech command corpus(1 second).
When I finished the first epoch, I have some problem in validation because of the stoi loss, the stft of the audio is ([257, 24]), but the function of stoi loss which in **speechbrain/nnet/loss/stoi.loss.py**, the parameter **N** is 30 (length of temporal envelope vectors), and line 189

`M = int(stft_pred.shape[-1] - (N - 1))`

24 - (30 - 1) = -5, so the error I get is 

**RuntimeError: Trying to create tensor with negative dimension -75: [-75, 30]**

Any help appreciated!

**Comment by jiaweiru** [comment]
Hi, I'm also experiencing this problem, and I think it's caused by voice segments that are too short in length in the dataset. They can't calculate stoi properly, you can solve this error by deleting these short voices.
